pkg/image/oci/registry_provider_test.go

```diff
@@ -0,0 +1,46 @@
+package oci
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_prepareReferenceOptions(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    image.RegistryOptions
+		expected []name.Option
+	}{
+		{
+			name:     "not InsecureUseHTTP",
+			input:    image.RegistryOptions{},
+			expected: nil,
+		},
+		{
+			name: "use InsecureUseHTTP",
+			input: image.RegistryOptions{
+				InsecureUseHTTP: true,
+			},
+			expected: []name.Option{name.Insecure},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			out := prepareReferenceOptions(&test.input)
+			assert.Equal(t, len(test.expected), len(out))
```

Comment:
Optional — in case you're interested, the `assert` package has a native length assertion. (But there's nothing wrong with the way you've written this, either.)

```suggestion
			assert.Len(t, out, len(test.expected))
```

Reply:
I typically prefer coming up with an equals check in tests instead of relying on some behavior defined in a library helper method, I think it makes things easier for everyone to follow who may not be familiar with said testing library, but this is just my preference.